silk/tests/openthread/ot_test_network_data_timeout.py

```diff
@@ -0,0 +1,338 @@
+        """
+        This function verifies that none of the nodes in `node_list` contains the on-mesh `prefix` associated with the
+        given `rloc16`.
+        """
+        for node in node_list:
+            prefixes = wpan_table_parser.parse_on_mesh_prefix_result(
+                node.get(wpan.WPAN_THREAD_ON_MESH_PREFIXES))
+
+            for p in prefixes:
+                if (p.prefix == prefix and p.origin == "ncp" and
+                        int(p.rloc16(), 0) == rloc16):
+                    print("Did find prefix {} with rloc16 {} on node {}".format(prefix, hex(rloc16), node))
+                    exit(1)
+
+    @testcase.test_method_decorator
+    def test01_pairing(self):
+        self.r1.whitelist_node(self.r2)
+        self.r2.whitelist_node(self.r1)
+
+        self.r2.whitelist_node(self.sc1)
+        self.sc1.whitelist_node(self.r2)
+
+        self.r1.form(self.network_data, "router")
+        self.r1.permit_join(3600)
+        self.wait_for_completion(self.device_list)
+
+        self.logger.info(self.r1.ip6_lla)
+        self.logger.info(self.r1.ip6_thread_ula)
+
+        self.network_data.xpanid = self.r1.xpanid
+        self.network_data.panid = self.r1.panid
+
+        self.r2.join(self.network_data, 'router')
+
+        self.sc1.join(self.network_data, "sleepy-end-device")
+        self.sc1.set_sleep_poll_interval(self.poll_interval)
+
+        self.wait_for_completion(self.device_list)
+
+        for _ in range(18):
+            node_type = self.r2.wpanctl('get', 'get '+wpan.WPAN_NODE_TYPE, 2).split('=')[1].strip()[1:-1]
+            self.logger.info(node_type == 'router')
+
+            if node_type == 'router':
+                self.logger.info('Matched!!!!!!!!!!!!!')
+                break
+            time.sleep(10)
+        else:
+            self.assertFalse(True, 'Router cannot get into router role after 180 seconds timeout')
+
+    @testcase.test_method_decorator
+    def test02_verify_prefix(self):
+        # Add 4 prefixes(`prefix1`, 'prefix2', `prefix3` and `common_prefix`) and
+        # verify they are present on all nodes with correct flags
+
+        # Add a unique prefix on each node.
+        self.r1.add_prefix(prefix1, on_mesh=True, preferred=True, stable=True)
+        self.r2.add_prefix(prefix2, on_mesh=True, preferred=True, stable=True)
+        self.sc1.add_prefix(prefix3, on_mesh=True, preferred=True, stable=True)
+
+        # Add `common_prefix` on all three nodes with different flags
+        self.r1.add_prefix(common_prefix, on_mesh=True, preferred=True, stable=False)
+        self.r2.add_prefix(common_prefix, on_mesh=True, preferred=True, stable=True)
+        self.sc1.add_prefix(common_prefix, on_mesh=True, preferred=False, stable=True)
+
+        # Wait for on-mesh prefixes to be updated
+        self.wait_for_completion(self.device_list)
+        time.sleep(60)
+
+        # Get all nodes rloc16
+        global r1_rloc16, r2_rloc16, sc1_rloc16
```

Comment:
Do we need global ?

Reply:
Not really. But these rloc16 values are required in subtests test02 and test03 . Thus defined them as global in test02 and used them in test03 too instead of defining them again for test03  hence avoiding redundant code.

Please let me know if there is any other way we can address this. Thanks in advance.

Reply:
A variable inside the class should be good.